sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -374,8 +380,46 @@ public void addIntegration(@NotNull Integration integration) {
    *
    * @param dsn the DSN
    */
-  public void setDsn(@Nullable String dsn) {
+  public void setDsn(final @Nullable String dsn) {
     this.dsn = dsn;
+
+    dsnHash = calculateHashedDsn(this.dsn);
+  }
+
+  private @Nullable String calculateHashedDsn(final @Nullable String dsn) {
+    if (dsn == null || dsn.isEmpty()) {
+      return null;
+    }
+
+    try {
+      // getInstance() method is called with algorithm SHA-1
+      final MessageDigest md = MessageDigest.getInstance("SHA-1");
+
+      // digest() method is called
+      // to calculate message digest of the input string
+      // returned as array of byte
+      final byte[] messageDigest = md.digest(dsn.getBytes(UTF_8));
+
+      // Convert byte array into signum representation
+      final BigInteger no = new BigInteger(1, messageDigest);
+
+      // Convert message digest into hex value
+      String hashtext = no.toString(16);
+
+      // Add preceding 0s to make it 32 bit
+      while (hashtext.length() < 32) {
+        hashtext = "0" + hashtext;
+      }
+
+      // return the HashText
+      return hashtext;
```

Comment:
I could make a StringBuilder

Reply:
yeah lets use the builder to avoid an allocation for each concatenation